mlflow/openai/__init__.py

```diff
@@ -639,34 +637,6 @@ def __init__(self, model):
         self.task = task
         self.api_config = _get_api_config()
         self.api_token = _OAITokenHolder(self.api_config.api_type)
-        # If the same parameter exists in self.model & self.api_config,
-        # we use the parameter from self.model
-        self.request_configs = {}
-        for x in ["api_base", "api_version", "api_type", "engine", "deployment_id"]:
-            if x in self.model:
-                self.request_configs[x] = self.model.pop(x)
-            elif value := getattr(self.api_config, x):
-                self.request_configs[x] = value
-
-        if self.request_configs.get("api_type") in ("azure", "azure_ad", "azuread"):
-            deployment_id = self.request_configs.get("deployment_id")
-            if self.request_configs.get("engine"):
-                # Avoid using both parameters as they serve the same purpose
```

Comment:
Does this validation covered by OAI client?

Reply:
> we don't manually construct parameters any more

We still construct them via `_get_api_config()` right?  I think the intention here is avoid passing `OPENAI_ENGINE` and `OPENAI_DEPLOYMENT_NAME` together, which can still happen.

So my question is whether OAI client can catch this conflict and raise an exception. Or is `OPENAI_ENGINE` fully deprecated and we can safely remove this validation?

Reply:
Discussed offline: `OPENAI_ENGINE` seems to be deprecated a while ago so we don't need to support it.

Reply:
Yes, engine param was replaced by "model" in OpenAI and by "deployment_id" in Azure.
https://github.com/openai/openai-python/discussions/742
https://learn.microsoft.com/en-us/azure/ai-services/openai/reference